From 44f6e48cd704d5cd2d047d3b77b4773a312efa2c Mon Sep 17 00:00:00 2001
From: Yelena Krivosheev <yelena@marvell.com>
Date: Mon, 26 Nov 2018 08:02:03 +0200
Subject: [PATCH 0806/1051] net: mvneta: fix user space port link change
 process

For US port don't reconfigure MAC and don't force
link down.

Change-Id: I7196e6c956035b8b8e41acc93df8b0a244650878
Signed-off-by: Yelena Krivosheev <yelena@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61302
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvneta.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvneta.c b/drivers/net/ethernet/marvell/mvneta.c
index 9158a375af3b..5abce2ea138f 100644
--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -3621,8 +3621,10 @@ static void mvneta_mac_link_down(struct net_device *ndev, unsigned int mode,
 	struct mvneta_port *pp = netdev_priv(ndev);
 	u32 val;
 
-	if (!pp->musdk_port)
-		mvneta_port_down(pp);
+	if (pp->musdk_port)
+		return;
+
+	mvneta_port_down(pp);
 
 	if (!phylink_autoneg_inband(mode)) {
 		val = mvreg_read(pp, MVNETA_GMAC_AUTONEG_CONFIG);
-- 
2.17.1

